rfds/rfd-job-worker-service.md

```diff
@@ -0,0 +1,381 @@
+- Error handling refinement
+- Performance optimization
+
+### Key Design Decisions
+
+**Authentication**: mTLS-only approach eliminates password management complexity while providing strong mutual authentication.
+
+**Output Streaming**: Event-driven capture with atomic buffer storage enables lock-free concurrent access. Moving pointer approach delivers complete history plus live data in unified streaming model. Each client maintains independent offset tracking for gap-free delivery without polling or busy-waiting.
+
+**User Isolation**: Per-user process isolation ensures users can only access jobs they created, with server-side enforcement at the API layer.
+
+**Concurrency**: Channel-based coordination with atomic operations avoids mutex complexity. Lock-free architecture prevents deadlocks while supporting unlimited concurrent streaming clients.
+
+
+## Security
+
+### Authentication & Authorization
+- **mTLS**: Mutual TLS for all client-server communication
+
+### TLS Configuration
+- **Cipher Selection**: ChaCha20-Poly1305 chosen over AES-GCM for timing-attack immunity and security robustness
+- **Performance Trade-off**: While AES-GCM offers 10-20% better performance on modern hardware with AES-NI, cipher overhead (microseconds) is negligible compared to process spawn times (100ms+) and network latency (10ms+)
+- **Security Priority**: Decision prioritizes security over premature optimization, eliminating entire classes of implementation vulnerabilities
+- **Certificate Management**: 15-day expiration for demo purposes, 4096-bit RSA keys with SHA-256 signatures, server CN=localhost, client CNs for user identification
+
+### Process Security
+- **Command Execution**: Separate command and args fields prevent shell injection attacks using direct process execution (`exec.Command(command, args...)`) instead of shell parsing
+- **Process Isolation**: Users cannot access other users' jobs, job ownership verified on all operations
+- **Runtime Environment**: Jobs run with server process privileges in controlled environment
+- **Resource Protection**: Output streaming limited to prevent resource exhaustion, no arbitrary code execution in server process
+
+## Go Library API Specification
+
+The core library provides a clean interface that the gRPC server uses for job management. This design separates business logic from transport concerns.
+
+### Core Interface
+
+```go
+// JobManager provides the main library interface
+type JobManager interface {
+    StartJob(ctx context.Context, cmd string, args []string, owner string) (string, error)
+    StopJob(ctx context.Context, jobID string, owner string) error
+    GetJobStatus(ctx context.Context, jobID string, owner string) (*JobStatus, error)
+    StreamOutput(ctx context.Context, jobID string, owner string) (<-chan OutputChunk, error)
+}
+
+// JobStatus represents current job state
+type JobStatus struct {
+    JobID     string
+    State     JobState
+    ExitCode  int32
+    StartTime time.Time
```

Comment:
While nice to have, for the sake of time, I'd recommend to skip time metadata and keep things as simple as posisble.